crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -123,6 +124,70 @@ impl Persistence {
             .map_err(DatabaseError)
     }
 
+    /// Save all valid solutions that participated in the competition for an
+    /// auction.
+    pub async fn save_solutions(
+        &self,
+        auction_id: domain::auction::Id,
+        solutions: &[domain::competition::Participant],
+    ) -> Result<(), DatabaseError> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["save_solutions"])
+            .start_timer();
+
+        let mut ex = self.postgres.pool.begin().await?;
+
+        database::solver_competition::save_solutions(
+            &mut ex,
+            auction_id,
+            &solutions
+                .iter()
+                .enumerate()
+                .map(|(uid, participant)| {
+                    let solution = Solution {
+                        uid: uid.try_into().context("uid overflow")?,
```

Comment:
The db docs made it seem like this `uid` is supposed to be globally unique, which I think is a nicer property than just having it be the index within one auction.

Reply:
`unique id of the proposed solution within a single auction` this is in the docs